tests/lambda_endpoints/test_batch_starter.py

```diff
@@ -263,8 +179,69 @@ def test_lambda_handler_multiple_events(session, mock_urlopen):
         assert mock_batch_client.submit_job.call_count == 2
 
 
+def test_lambda_handler_ancillary_event(
+    session,
+    mock_urlopen: unittest.mock.MagicMock,
+):
+    """Tests ``lambda_handler`` function when triggerd by an ancillary file."""
+    _populate_file_catalog(session)
+    events = {
+        "Records": [
+            {
+                "body": '{"detail": '
+                '{"object": {"key": '
+                '"imap_swe_l1b-in-flight-cal_20240101_v001.cdf"}}'
+                "}"
+            }
+        ]
+    }
+
+    context = {"context": "sample_context"}
+    with patch.object(batch_starter, "BATCH_CLIENT", Mock()) as mock_batch_client:
+        lambda_handler(events, context)
+        # There should be two different jobs submitted for one swe l1b ancillary file
+        assert mock_batch_client.submit_job.call_count == 2
+        # Assert_called_with only works on the last call
+        # Check that the last call is what we expect with the corrected
+        ancillary_in = AncillaryInput(
+            "imap_swe_l1b-in-flight-cal_20230101_v001.cdf",
+            "imap_swe_l1b-in-flight-cal_20231231_20240102_v002.cdf",
+        )
+        science_in = ScienceInput(
+            "imap_swe_l1a_sci_20240103_v001.cdf",
+        )
+        dependencies = ProcessingInputCollection(science_in, ancillary_in)
+        mock_batch_client.submit_job.assert_called_with(
+            jobName="swe-l1b-sci-job-2",
+            jobQueue="ProcessingJobQueue",
+            jobDefinition="ProcessingJob-swe",
+            containerOverrides={
+                "command": [
+                    "--instrument",
+                    "swe",
+                    "--data-level",
+                    "l1b",
+                    "--descriptor",
+                    "sci",
+                    "--start-date",
+                    "20240103",
+                    "--version",
+                    "v001",
+                    "--dependency",
+                    dependencies.serialize(),
+                    "--upload-to-sdc",
+                ]
+            },
+        )
+        # Submit a second job with the same file as input which will try to kick
+        # off a duplicate job. We expect the submit_job method to not be called
+        # so make sure it is still only called three times from our previous iteration.
+        lambda_handler(events, context)
+        assert mock_batch_client.submit_job.call_count == 2
```

Comment:
Might want to reset this to a value you control and/or add assert_not_called() after this?
```suggestion
        # so make sure it is still only called two times from our previous iteration.
        mock_batch_client.submit_job.call_count == 0
        lambda_handler(events, context)
        assert mock_batch_client.submit_job.call_count == 0
```